Packs/GenericWebhook/Integrations/GenericWebhook/GenericWebhook.py

```diff
@@ -0,0 +1,149 @@
+from pydantic import BaseModel
+
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+sample_events_to_store = deque(maxlen=20)  # type: ignore[var-annotated]
+
+
+class Incident(BaseModel):
+    name: Optional[str] = None
+    type: Optional[str] = None
+    occurred: Optional[str] = None
+    raw_json: Optional[Dict] = None
+
+
+app = FastAPI(logger=DebugLogger(), docs_url=None, redoc_url=None, openapi_url=None)
+
+basic_auth = HTTPBasic(auto_error=False)
+token_auth = APIKeyHeader(auto_error=False, name='Authorization')
+
+
+@app.post('/')
+async def handle_post(
+        incident: Incident,
+        request: Request,
+        credentials: HTTPBasicCredentials = Depends(basic_auth),
+        token: APIKey = Depends(token_auth)
+):
+    credentials_param = demisto.params().get('credentials')
+    if credentials_param:
+        username = credentials_param.get('identifier', '')
+        password = credentials_param.get('password', '')
+        auth_failed = False
+        if username.startswith('_header'):
+            header_name = username.split(':')[1]
+            token_auth.model.name = header_name
+            if not token or not compare_digest(token, password):
+                auth_failed = True
+        elif not (compare_digest(credentials.username, username) and compare_digest(credentials.password, password)):
+            auth_failed = True
+        if auth_failed:
+            demisto.debug(f'Authorization failed - request headers {request.headers}')
+            return Response(status_code=status.HTTP_401_UNAUTHORIZED, content='Authorization failed.')
+
+    raw_json = incident.raw_json or await request.json()
+    incident = {
+        'name': incident.name or 'Generic webhook triggered incident',
+        'type': incident.type or demisto.params().get('incidentType'),
+        'occurred': incident.occurred,
+        'rawJSON': json.dumps(raw_json)
+    }
+
+    if demisto.params().get('store_samples'):
+        try:
+            sample_events_to_store.append(incident)
+            integration_context = get_integration_context()
+            sample_events = deque(json.loads(integration_context.get('sample_events', '[]')), maxlen=20)
+            sample_events += sample_events_to_store
+            integration_context['sample_events'] = list(sample_events)
+            set_to_integration_context_with_retries(integration_context)
+        except Exception as e:
+            demisto.error(f'Failed storing sample events - {e}')
+
+    return demisto.createIncidents([incident])
+
+
+def fetch_samples() -> None:
+    """Extracts sample events stored in the integration context and returns them as incidents
+
+    Returns:
+        None: No data returned.
+    """
+    integration_context = get_integration_context()
+    sample_events = json.loads(integration_context.get('sample_events', '[]'))
+    incidents = [{'rawJSON': json.dumps(event)} for event in sample_events]
```

Comment:
don't you miss 'name' for the incident?

Reply:
not needed
thats what it looks like:
![image](https://user-images.githubusercontent.com/31018228/97102315-0725c080-16ad-11eb-8723-3b6679af0caf.png)

Reply:
but we should only show in the samples the rawJSON. THis is how pull from instance shows the samples in regular-fetch incidents

Reply:
i've changed it to return incident object, which includes the name, rawjson, type and occurred fields, as is without modifications - like we do in regular fetch